traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/config/ConfigHandler.java

```diff
@@ -454,6 +476,31 @@ private void parseCacheConfig(final JsonNode contentServers, final CacheRegister
 		return deliveryServiceMap;
 	}
 
+	private void parseTopologyConfig(final JsonNode allTopologies, final Map<String, DeliveryService> deliveryServiceMap, final CacheRegister cacheRegister) {
+		final Map<String, List<String>> topologyMap = new HashMap<>();
+		allTopologies.fieldNames().forEachRemaining((String topologyName) -> {
+			final List<String> nodes = new ArrayList<>();
+			allTopologies.get(topologyName).get("nodes").forEach((JsonNode cache) -> nodes.add(cache.textValue()));
+			topologyMap.put(topologyName, nodes);
+		});
+
+		deliveryServiceMap.forEach((xmlId, ds) -> {
+			final List<DeliveryServiceReference> dsReferences = new ArrayList<>();
+			try {
+				dsReferences.add(new DeliveryServiceReference(ds.getId(), ds.getDomain()));
```

Comment:
So I think `ds.getDomain()` is wrong here. Looking at how this is done for non-Topology-based DSes: https://github.com/apache/trafficcontrol/blob/master/traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/config/ConfigHandler.java#L393
in the CRConfig those names are not just the DS domain -- they're the full cache hostnames (like in the example delivery URLs) plus any CNAME aliases:
```
  "contentServers": {
    "edge-01": {
      "deliveryServices": {
        "foo": [
          "edge-01.foo.mycdn.example.com",
          "blah.example.com",
          "bar.someotherdomain.example.com"
        ],
```

I attempted to curl a topology-based DS, but was 302'd to just the DS domain (e.g. foo.mycdn.example.com) rather than the actual cache (e.g. edge-01.mycdn.example.com) as expected. I think we may need to basically replicate the functionality of TO's CRConfig generation for those hostname arrays in TR for topology-based DSes. I think we should have enough information in CRConfig to generate that data locally, since I think we may be able to use the  `matchsets` array on the DS objects.

Here is how it's done in TO: https://github.com/apache/trafficcontrol/blob/master/traffic_ops/traffic_ops_golang/crconfig/servers.go#L257

Reply:
Data generated locally in c6b7ed474d

Reply:
So I think that covers the "normal" use case, where a delivery service doesn't have any "vanity URLs" -- created via adding another HOST_REGEXP with set > 0, e.g. foo.example.com -- but does it handle the Vanity URL use case?

Say my "normal" URL is cdn.myds.cdn.example.com (xml ID is "myds"), if you add another HOST_REGEXP value "foo.example.com" with order = 1, "foo.example.com" becomes the Vanity URL. This works by having the foo.example.com CNAME to cdn.myds.cdn.example.com. So the `Host` client header that TR sees is "foo.example.com", and it needs to match that URL to the "myds" delivery service somehow.

Looking closely at https://github.com/apache/trafficcontrol/blob/master/traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/config/ConfigHandler.java#L398-L408 though, it looks like really pertains to `statMap` handling, which makes me think we'll probably _route_ the request fine, but it might not be counted for the right DS in that `statsMap`? That `statsMap` stuff should be http://localhost:3333/crs/stats by the way.

Reply:
As of af6c35ecd, all delivery service URLs are considered when parsing the topologies config (including initialization of stat tracking)